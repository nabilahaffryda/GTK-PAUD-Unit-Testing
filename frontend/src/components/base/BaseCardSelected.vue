<template>
  <v-card outlined :disabled="(selected && selected.disable) || false">
    <v-list-item @click="onClick">
      <v-list-item-content class="ma-0 pa-0">
        <template v-if="selected && selected !== null">
          <list-avatar
            :title="(selected && selected.nama) || ''"
            :subtitle="(selected && selected.desc) || ''"
            class="py-0"
          >
            <v-icon>{{ selected.icon }}</v-icon>
          </list-avatar>
        </template>
        <template v-else>
          <v-list-item-title class="text-capitalize" v-html="title"></v-list-item-title>
        </template>
      </v-list-item-content>
      <v-list-item-action>
        <v-btn icon>
          <v-icon color="grey lighten-1"> mdi-arrow-down-drop-circle-outline </v-icon>
        </v-btn>
      </v-list-item-action>
    </v-list-item>
  </v-card>
</template>

<script>
export default {
  name: 'SelectCard',
  props: {
    title: {
      type: String,
      default: 'Pilih selector',
    },
    selected: {
      type: Object,
    },
  },
  methods: {
    onClick() {
      this.$emit('click');
    },
  },
};
</script>

<style scoped></style>
