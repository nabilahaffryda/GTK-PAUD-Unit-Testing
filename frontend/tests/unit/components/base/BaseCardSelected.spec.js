import { mount, createLocalVue } from "@vue/test-utils";
import BaseCardSelected from "@/components/base/BaseCardSelected.vue";
import Vuetify from 'vuetify';
import Vue from 'vue';

Vue.use(Vuetify)
const localVue = createLocalVue();

describe('BaseCardSelected.vue', () => {
    let vuetify
    beforeEach(() => {
        vuetify = new Vuetify()
    })
    test('should show card selected', () => {
        const VCard = {
            props: ['title', 'subtitle', 'desc',
                'icon', 'color', 'deepColor',
                'akses', 'disable', 'to',
                'href', 'target', 'action'],
            template: '<div><slot name="button" /></div>'
        }

        const wrapper = mount(BaseCardSelected, {
            localVue,
            vuetify,
            stubs: {
                VCard
            }
        });
        expect(wrapper).toMatchSnapshot();
    })
})
