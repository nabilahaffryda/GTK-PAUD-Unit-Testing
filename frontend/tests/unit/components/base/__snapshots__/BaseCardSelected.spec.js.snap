// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BaseCardSelected.vue should show card selected 1`] = `<div outlined=""></div>`;
